src/os/ObjectStore.h

```diff
@@ -431,6 +431,40 @@ class ObjectStore {
         largest_data_off_in_tbl(0),
 	fadvise_flags(0) { }
 
+      // as a plain-old-data type, the move operations are classified as
+      // 'trivial', which means that the compiler would generate code identical
+      // to the copy operations (i.e. with std::memmove). we want the source
+      // object to end up zero-initialized, however, so we have to override the
+      // default implementations to do that
+      TransactionData(TransactionData&& other) :
+        ops(other.ops),
```

Comment:
Not sure, do we need explicate std::move? otherwise it should be a copy construct?